### PR TITLE
Fixes #34178 - add generic repository testing

### PR DESCRIPTION
**Pull request by jlsherrill**
and enable an ostree test

#### What are the changes introduced in this pull request?
Adds the ability to have tests as part of a repository type without having to write any tests.

#### Considerations taken when implementing this change?
This adds some testing code (declaration of fixture repo) into the production repository_type definition.  To me this is okay, but wanted to get opinions.

#### What are the testing steps for this pull request?
Tests pass, you like the approach

### CODE REVIEW

**Comment by jlsherrill** [comment]
I'll add repository upload support if this is a good approach

**Review by rverdile — COMMENTED** [review]
I like this idea. Running the test gives me an error. Is it supposed to work yet?

**Comment by jlsherrill** [comment]
I don't remember! let me rebase this and test locally ;)

File: test/services/katello/pulp3/repository_integration_test.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          ForemanTasks.sync_task(
+            ::Actions::Pulp3::Orchestration::Repository::Delete, @repo, @primary)
+        end
+      end
+
+      def test_sync
+        @repo.update(publication_href: nil, version_href: nil)
+        ForemanTasks.sync_task(::Actions::Katello::Repository::Sync, @repo)
+        @repo.reload
+        refute_equal @repo.version_href, @repo_version_href
```
Comment (rverdile) [inline test/services/katello/pulp3/repository_integration_test.rb:41]: does `@repo_version_href` exist?

Comment (jlsherrill) [reply test/services/katello/pulp3/repository_integration_test.rb:41]: i don't think !  this line isn't really doing much, i'll remove it

Comment (jlsherrill) [reply test/services/katello/pulp3/repository_integration_test.rb:41]: good eye!

**Comment by rverdile** [comment]
are you done with this or did you want to add more tests to it?

File: test/services/katello/pulp3/repository_integration_test.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      def test_upload_files
+        repo_type.content_types.each do |content_type|
+          next unless content_type.test_upload_path
+
+          to_upload = File.join(Katello::Engine.root, content_type.test_upload_path)
+          @file = {path: to_upload, filename: File.basename(content_type.test_upload_path)}
+
+          assert_equal 0, @repo.content_counts[content_type.label]
+          VCR.use_cassette(cassette_name + '_binary', :match_requests_on => [:method, :path, :params]) do
+            #action_result = ForemanTasks.sync_task(::Actions::Pulp3::Orchestration::Repository::UploadContent, @repo, @primary, @file, content_type.label)
```
[the remark this answers is not in the document]
Comment (jlsherrill) [reply test/services/katello/pulp3/repository_integration_test.rb:64]: oh, it should have been deleted, i replaced it with the line below (which also does the indexing)

**Comment by jlsherrill** [comment]
updated!

**Review by rverdile — APPROVED** [review]
lgtm! Tests work and so did the `save_artifact.rb` change.

**Comment by jlsherrill** [comment]
[test katello]

**Pull request merged**